src/term/cell.rs

```diff
@@ -82,6 +84,8 @@ impl Cell {
             c: c.into(),
             bg: bg,
             fg: fg,
+            undercursorbg: bg,
```

Comment:
These properties need to go somewhere else other than `Cell`. The `Grid` is a collection of up to ~40,000 cells, and any size increase on cell has a big impact.

I would consider putting them on the `RenderableCellsIter`.